lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -66,11 +63,12 @@ def query_api(
 
         return query_api
 
-    def token(self, user: str) -> str:
-        roles = None
-        if user == "test_admin":
-            roles = ["ADMIN"]
-        return generate_token(username=user, pbench_client_roles=roles)
+    def token(self, pbench_admin_token, user: str) -> str:
+        return (
+            pbench_admin_token
+            if user == admin_username
+            else generate_token(username=user)
+        )
```

Comment:
Consider implementing this as a fixture in `lib/pbench/test/unit/server/conftest.py` instead of as a class method:
```
@pytest.fixture()
def get_token(pbench_admin_token):
    def token(user: str) -> str:
        return (
            pbench_admin_token
            if user == admin_username
            else generate_token(username=user)
        )
    return token
```
So, then, line 54 becomes
```
                token = get_token(username)
```
and you can use `get_token` in the other `query_as()` functions as well, and avoid the duplication.

And, if you like, you can define `get_token()` like this:
```
@pytest.fixture()
def get_token(pbench_admin_token):
    return lambda user: (
        pbench_admin_token if user == admin_username else generate_token(username=user)
    )
```